app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_history = {}
+
+    def wrapper(*args: Any) -> Callable:
+        nonlocal cache_history
+        cache_key = args
+
+        if cache_key in cache_history:
+            print("Getting from cache")
+            return cache_history[cache_key]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_history[cache_key] = result
+            return result
+    return wrapper
```

Comment:
You should check both `args` and `kwargs` because the function might accept the same `args` but different `kwargs`. Additionally, I recommend checking the function name since there could be two different functions that accept the same arguments.

Reply:
I fixed the errors you pointed out